source/source.go

```diff
@@ -0,0 +1,186 @@
+// Copyright © 2022 Meroxa, Inc. & Yalantis
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package source
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/conduitio-labs/conduit-connector-clickhouse/config"
+	"github.com/conduitio-labs/conduit-connector-clickhouse/source/iterator"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/multierr"
+
+	// Go driver for ClickHouse.
+	_ "github.com/ClickHouse/clickhouse-go/v2"
+)
+
+// Iterator interface.
+type Iterator interface {
+	HasNext(context.Context) (bool, error)
+	Next(context.Context) (sdk.Record, error)
+	Stop() error
+}
+
+// Source connector.
+type Source struct {
+	sdk.UnimplementedSource
+
+	config   config.Source
+	db       *sqlx.DB
+	iterator Iterator
+}
+
+// NewSource initialises a new source.
+func NewSource() sdk.Source {
+	return sdk.SourceWithMiddleware(&Source{}, sdk.DefaultSourceMiddleware()...)
+}
+
+// Parameters returns a map of named Parameters that describe how to configure the Source.
+func (s *Source) Parameters() map[string]sdk.Parameter {
+	return map[string]sdk.Parameter{
+		config.URL: {
+			Default:     "",
+			Required:    true,
+			Description: "The connection string to connect to ClickHouse database.",
+		},
+		config.Table: {
+			Default:     "",
+			Required:    true,
+			Description: "The table name of the table in ClickHouse that the connector should write to, by default.",
+		},
+		config.KeyColumns: {
+			Default:     "",
+			Required:    true,
+			Description: "The names of the columns to build the record.Key, separated by commas.",
```

Comment:
I think we need a clearer description of this param, especially since it's required

Reply:
I've added some additional information about how we build this field.
Could you check it?